linkschkr_test.go

```diff
@@ -0,0 +1,69 @@
+package linkschkr
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestCrawler(t *testing.T) {
+	t.Parallel()
+
+	nWorkers := 1000 // just for fun
+	work := make(chan *Work)
+	results := make(chan *Result)
+
+	for n := 0; n < nWorkers; n++ {
+		go Crawler(n, work)
+	}
+
+	body := `<html>
+	<head>
+		<title>For the love of Go!</title>
+	</head>
+	<body>
+		Hi folks,
+
+		See docs <a href="/docs">Go docs</a>. Go is awesome!
+		ajkwerjakwerjkaewrhjakwehrjkawe
+		aelwrkhakelrhalkewrhkleawlhrwaklerhalkewrhalewrk
+
+		Visit <a href="/live">our live</a>.
+		aewrjaewporawe
+
+		awerjaioewrhaewoi
+		a href="testintentional.com">trying to get you in trouble</a>
+	</body>
+</html>`
+
+	ts := httptest.NewServer(
```

Comment:
It's a little more work, but let's use a `NewTLSServer` here instead. We wouldn't want to find that our carefully-tested client code doesn't work with TLS servers for some reason! And, in general, I think it's wise to avoid ever writing non-HTTPS servers or clients.

Reply:
based on what we've been discussing, to test the http package shouldn't be part of my tests, should it?